packages/miniflare/src/plugins/core/modules.ts

```diff
@@ -133,6 +133,16 @@ function moduleName(modulesRoot: string, modulePath: string) {
   // Module names should always use `/` as the separator
   return path.sep === "\\" ? name.replaceAll("\\", "/") : name;
 }
+export function withSourceURL(script: string, scriptPath: string): string {
+  let scriptURL: URL | string = scriptPath;
+  if (maybeGetStringScriptPathIndex(scriptPath) === undefined) {
+    scriptURL = pathToFileURL(scriptPath);
+  }
+  let sourceURL = `//# sourceURL=${scriptURL}\n`;
+  // Make sure `//# sourceURL` comment is on its own line
+  if (!script.endsWith("\n")) sourceURL = "\n" + sourceURL;
```

Comment:
can the preceding `\n` be added unconditionally?

```suggestion
  const sourceURL = `\n//# sourceURL=${scriptURL}\n`;
```